### PR TITLE
[sync-commit.py] Stop syncing SVN revision after a certain revision

### PR DESCRIPTION
**Pull request by facetothefate**
#### e033cd052146ad8b8ccae8221760d8c0be8e5ac5
<pre>
[sync-commit.py] Stop syncing SVN revision after a certain revision

Patch by Zhifei Fang &lt;facetothefate@gmail.com&gt; on 2022-06-17
Reviewed by Dewei Zhu.

* Websites/perf.webkit.org/tools/sync-commits.py:
(load_repository):
(GitRepository.__init__):
(GitRepository.fetch_next_commit):
(GitRepository._svn_revision_from_git_hash):
(GitRepository._git_hash_from_svn_revision_hash_mixed):
(GitRepository._revision_from_tokens):
(GitRepository._git_hash_from_svn_revision): Deleted.

Canonical link: <a href="https://commits.webkit.org/251658@main">https://commits.webkit.org/251658@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295653">https://svn.webkit.org/repository/webkit/trunk@295653</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Websites/perf.webkit.org/tools/sync-commits.py [hunk 1]
```diff
@@ -49,7 +53,8 @@ def load_repository(repository):
         return GitRepository(
             name=repository['name'], git_url=repository['url'], git_checkout=repository['gitCheckout'],
             git_branch=repository.get('branch'), report_revision_identifier_in_commit_msg=repository.get('reportRevisionIdentifier'),
-            report_svn_revison=repository.get('reportSVNRevision'))
+            report_svn_revison=repository.get('reportSVNRevision'),
+            revison_for_stop_report_svn_revision=repository.get('revisionForStopReportSVNRevision'))
```
Comment (JonWBedard) [inline Websites/perf.webkit.org/tools/sync-commits.py]: Maybe `last_revision`?

File: Websites/perf.webkit.org/tools/sync-commits.py [hunk 2]
```diff
@@ -18,7 +19,10 @@
 
 # There are some buggy commit messages:
 # Canonical link: https://commits.webkit.org/https://commits.webkit.org/232477@main
-REVISION_IDENTIFIER_RE = re.compile(r'Canonical link: (https\://commits\.webkit\.org/)+(?P<revision_identifier>\d+@[\w\.\-]+)\n')
+REVISION_IDENTIFIER_IN_MSG_RE = re.compile(r'Canonical link: (https\://commits\.webkit\.org/)+(?P<revision_identifier>\d+\.@[\w\.\-]+)\n')
```
Comment (JonWBedard) [inline Websites/perf.webkit.org/tools/sync-commits.py]: Should be `\d+\.?\d*...` to match branch commits.

File: Websites/perf.webkit.org/tools/sync-commits.py (hunk 1 above)
Comment (dewei-zhu) [reply Websites/perf.webkit.org/tools/sync-commits.py]: Or `max_revision` ?

File: Websites/perf.webkit.org/tools/sync-commits.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/python
 
 import argparse
+from email import message
```
Comment (dewei-zhu) [inline Websites/perf.webkit.org/tools/sync-commits.py]: Is this used anywhere?

**Review by dewei-zhu — APPROVED** [review]
r=me with comment

File: Websites/perf.webkit.org/tools/sync-commits.py (hunk 3 above)
Comment (facetothefate) [reply Websites/perf.webkit.org/tools/sync-commits.py]: No, I guess my editor auto import

**Pull request merged**